src/Filters/coefficients.jl

```diff
@@ -317,9 +322,9 @@ function split_real_complex(x::Vector{T}) where T
 
     c = T[]
     r = typeof(real(zero(T)))[]
-    for k in sort!(collect(keys(d)), by=x -> (real(x), imag(x)))
+    for k in keys(d)
```

Comment:
Unrelated clean-up: The order in which this loop proceeds doesn't matter and the results are sorted afterwards anyway (and by distance to the unit circle, not by this canonical order).